### PR TITLE
#1046 Automatic Naming of Pricelist Version Name

### PR DESCRIPTION
**Pull request by metas-rc**
Defined identifiers for M_PriceList and M_PriceListVersion.
Made sure the dates are displayed without time.
FRESH-1549 https://github.com/metasfresh/metasfresh/issues/1046

### CODE REVIEW

File: de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/MLookupFactory.java [hunk 1]
```diff
@@ -690,7 +690,14 @@ private static String buildDisplayColumnSql(final ILookupDisplayColumn ldc, fina
 		// date
 		else if (DisplayType.isDate(ldc.getDisplayType()))
 		{
-			return DB.TO_CHAR(columnSQL, ldc.getDisplayType());
+
```
Comment (teosarca) [inline de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/MLookupFactory.java]: i think you shall do that only if DisplayType.Date.
For all others keep as it was.
NOTE: DIsplayType.isDate is returning true for Date, DateTIme and Time

File: de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/45-de.metas.pricing/5457370_gh1046_PriceListVersion_Identifier.sql [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+;
+
+-- 01.03.2017 14:32
+-- URL zum Konzept
+INSERT INTO AD_Table (AccessLevel,ACTriggerLength,AD_Client_ID,AD_Org_ID,AD_Table_ID,CopyColumnsFromTable,Created,CreatedBy,EntityType,ImportTable,IsActive,IsAutocomplete,IsChangeLog,IsDeleteable,IsDLM,IsHighVolume,IsSecurityEnabled,IsView,LoadSeq,Name,ReplicationType,TableName,Updated,UpdatedBy) VALUES ('3',0,0,0,540803,'N',TO_TIMESTAMP('2017-03-01 14:32:36','YYYY-MM-DD HH24:MI:SS'),100,'U','N','Y','N','N','Y','N','N','N','N',0,'testRC1Mar','L','testRC1Mar',TO_TIMESTAMP('2017-03-01 14:32:36','YYYY-MM-DD HH24:MI:SS'),100)
```
Comment (teosarca) [inline de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/45-de.metas.pricing/5457370_gh1046_PriceListVersion_Identifier.sql]: i think that shall not be here

File: de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/45-de.metas.pricing/5457370_gh1046_PriceListVersion_Identifier.sql [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+;
+
+-- 01.03.2017 14:32
+-- URL zum Konzept
+INSERT INTO AD_Table_Trl (AD_Language,AD_Table_ID, Name, IsTranslated,AD_Client_ID,AD_Org_ID,Created,Createdby,Updated,UpdatedBy) SELECT l.AD_Language,t.AD_Table_ID, t.Name, 'N',t.AD_Client_ID,t.AD_Org_ID,t.Created,t.Createdby,t.Updated,t.UpdatedBy FROM AD_Language l, AD_Table t WHERE l.IsActive='Y' AND l.IsSystemLanguage='Y' AND l.IsBaseLanguage='N' AND t.AD_Table_ID=540803 AND NOT EXISTS (SELECT * FROM AD_Table_Trl tt WHERE tt.AD_Language=l.AD_Language AND tt.AD_Table_ID=t.AD_Table_ID)
```
Comment (teosarca) [inline de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/45-de.metas.pricing/5457370_gh1046_PriceListVersion_Identifier.sql]: i think that shall not be here

File: de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/45-de.metas.pricing/5457370_gh1046_PriceListVersion_Identifier.sql [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+;
+
+-- 01.03.2017 14:52
+-- URL zum Konzept
+UPDATE AD_Element_Trl SET UpdatedBy=100,Updated=TO_TIMESTAMP('2017-03-01 14:52:07','YYYY-MM-DD HH24:MI:SS'),Name='testRC1Martran' WHERE AD_Element_ID=543288 AND AD_Language='fr_CH'
+;
+
+-- 01.03.2017 14:52
+-- URL zum Konzept
+UPDATE AD_Element_Trl SET UpdatedBy=100,Updated=TO_TIMESTAMP('2017-03-01 14:52:55','YYYY-MM-DD HH24:MI:SS'),Name='testRC1Martrans' WHERE AD_Element_ID=543288 AND AD_Language='fr_CH'
```
Comment (teosarca) [inline de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/45-de.metas.pricing/5457370_gh1046_PriceListVersion_Identifier.sql]: .... until here, i think that shall not be here

File: de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/45-de.metas.pricing/5457370_gh1046_PriceListVersion_Identifier.sql [hunk 5]
```diff
@@ -0,0 +1,120 @@
+-- 01.03.2017 14:31
+-- URL zum Konzept
+INSERT INTO AD_Element (AD_Client_ID,AD_Element_ID,AD_Org_ID,ColumnName,Created,CreatedBy,EntityType,IsActive,Name,PrintName,Updated,UpdatedBy) VALUES (0,543288,0,'testRC1Mar',TO_TIMESTAMP('2017-03-01 14:31:48','YYYY-MM-DD HH24:MI:SS'),100,'U','Y','testRC1Mar','testRC1Mar',TO_TIMESTAMP('2017-03-01 14:31:48','YYYY-MM-DD HH24:MI:SS'),100)
```
Comment (teosarca) [inline de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/45-de.metas.pricing/5457370_gh1046_PriceListVersion_Identifier.sql]: from here... i think that shall not be here

**Comment by teosarca** [comment]
also make sure there is an unique index on M_PriceList_Version(M_PriceList_ID, ValidFrom)

File: de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/MLookupFactory.java (hunk 1 above)
Comment (metas-rc) [inline de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/MLookupFactory.java]: Fixed.

File: de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/45-de.metas.pricing/5457370_gh1046_PriceListVersion_Identifier.sql (hunk 5 above)
Comment (metas-rc) [inline de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/45-de.metas.pricing/5457370_gh1046_PriceListVersion_Identifier.sql]: Right. Those were some leftovers from my tests. Sorry. for the mess. 
Fixed.

**Pull request merged**